### PR TITLE
VB-11: add sms sender service

### PR DESCRIPTION
**Pull request by dannleed**

### CODE REVIEW

File: packages/backend/src/services/sms/sms-sender.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   * @param {string} id - The id of the message to get.
+   * @returns The result of the getById method from the providerService.
+   */
+  public async getById(id: string): Promise<object> {
+    try {
+      const result = await this.providerService.getById(id);
+
+      return result;
+    } catch (error) {
+      console.error(error);
```
Comment (Nightlngale) [inline packages/backend/src/services/sms/sms-sender.ts:48]: use our "logger"

File: packages/backend/src/services/sms/twilio-sms-provider.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  authToken: string;
+  messagingServiceSid: string;
+}
+
+export class TwilioService implements ISMSSenderService, ITwilioOptions {
+  public authToken: ITwilioOptions['authToken'];
+
+  public accountSid: ITwilioOptions['accountSid'];
+
+  public messagingServiceSid: ITwilioOptions['messagingServiceSid'];
```
Comment (Nightlngale) [inline packages/backend/src/services/sms/twilio-sms-provider.ts:15]: public?

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/services/sms/sms-sender.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export interface ISMSSenderService {
+  send(phone: string, message: string): Promise<object>;
+  getById(id: string): Promise<object>;
+}
+
+class SMSSenderService implements ISMSSenderService {
+  private providerService: ISMSSenderService;
+
+  constructor() {
+    this.providerService = new TwilioService({
```
Comment (Vitalii4as) [inline packages/backend/src/services/sms/sms-sender.ts:14]: Create file `src/providers/sms/TwilioSMSProvider.ts` and move TwilioService there (of course rename class to TwilioSMSProvider). Create interface ISMSProvider and implement TwilioService from it. Recieve TwilioSMSProvider in constructor. Initialize and pass TwillioSMSProvider in `initServices`

**Review by Nightlngale — APPROVED** [review]

File: packages/backend/src/services/sms/sms-sender.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { log } from '@helpers';
+import type { ISMSProvider } from '@types';
+import type { ISMSSenderService } from '@types';
+
+class SMSSenderService implements ISMSSenderService {
+  private providerService: ISMSSenderService;
```
Comment (Vitalii4as) [inline packages/backend/src/services/sms/sms-sender.ts:6]: `private providerService: ISMSProvider;`

File: packages/backend/src/services/sms/sms-sender.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    this.providerService = providerService;
+  }
+
+  /**
+   * It sends a message to a phone number
+   * @param {string} phone - The phone number to send the message to.
+   * @param {string} message - The message you want to send.
+   * @returns The result of the send method from the provider service.
+   */
+  public async send(phone: string, message: string): Promise<object> {
```
Comment (Vitalii4as) [inline packages/backend/src/services/sms/sms-sender.ts:18]: Return just `Promise<Boolean>` - success or not

File: packages/backend/src/providers/sms/TwilioSMSProvider.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { getEnv, log } from '@helpers';
+import { Twilio } from 'twilio';
+import type { ISMSProvider } from '@types';
+
+export class TwilioSMSService implements ISMSProvider {
```
Comment (Vitalii4as) [inline packages/backend/src/providers/sms/TwilioSMSProvider.ts:5]: Rename to `TwilioSMSProvider`

**Review by Vitalii4as — CHANGES_REQUESTED** [review]

**Review by Vitalii4as — APPROVED** [review]

**Pull request merged**